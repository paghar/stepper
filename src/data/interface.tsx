import {gender,condation} from "./enum";

export interface IUser {
    id:string,
    name:string, 
    phone:string,   
    email:string,
    gender:gender,
    brithDay:string,
    address:string,
    height:string,
    weight:string,
    spokenLanguage:[],
    condation:condation
}

export interface IUserState{
    users:IUser,
    formStep: number 
}

export interface IStepItems {
    id:number,
    text:string,
}