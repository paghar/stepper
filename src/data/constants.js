export const stepItems = [
    {id:1, text:"Step one"},
    {id:2, text:"Step two"},
    {id:3, text:"Step three"},
    {id:4, text:"Step four"},
];

export const stepperNumber = 4;

export const Gender = [
    "Select Gender","Male","Female","Other"
];

export const SpokenLanguage = ["German","Franch","Spanish","Italian"];

export const condition = ["Yes", "No"];
