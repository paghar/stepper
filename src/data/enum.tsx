export enum gender {    
    Male="Male",
    Female="Female",
    Other="Other"
}

export enum spokenLanguage {
    German = "German",
    Franch = "Franch",
    Spanish = "Spanish",
    Italian = "Italian",
}

export enum condation {
    Yes = "Yes",
    No = "No"
}

export enum numbers {
    One = "1",
    Two = "2",
    Tree = "3",
    Four = "4",
}
