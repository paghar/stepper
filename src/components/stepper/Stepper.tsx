import React from "react";
import {IStepItems} from "../../data/interface";

interface IProps {
    stepprNumber:number,
    currentStep:number,
    stepItems:IStepItems[]
}

const Stepper = (props:IProps) => {

    const passedStepItem=(item:IStepItems)=>{
        return (
            <>
                <span className="h-10 w-10 rounded-full bg-blue-600 text-center text-[14px] font-bold leading-10 text-white">
                    <svg
                        xmlns="http://www.w3.org/2000/svg"
                        className="h-9 w-9"
                        viewBox="0 0 20 20"
                        fill="currentColor"
                    >
                        <path
                            fillRule="evenodd"
                            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                            clipRule="evenodd"
                        />
                    </svg>
                </span>           
                <span className="ml-4 text-base font-medium text-gray-900">{item.text}</span>
            </>
        );
    };

    const currentStepItem=(item:IStepItems)=>{
        return (
            <>
                <span className="h-10 w-10  border-2 rounded-full border-blue-600	 text-center text-[14px] font-bold leading-10 text-blue-600">
                    {item.id}
                </span>           
                <span className="ml-4 text-base font-medium text-blue-600">{item.text}</span>
            </>
        );
    };

    const upCommingStepItem=(item:IStepItems)=>{
        return (
            <>
                <span className="h-10 w-10  border-2 rounded-full  border-gray-400	 text-center text-[14px] font-bold leading-10 text-bule-600">
                    {item.id}
                </span>           
                <span className="ml-4 text-base font-medium text-bule-400">{item.text}</span>
            </>
        );
    };

    const stepItem= (item:IStepItems) =>{
        if (item.id === props.currentStep)
            return currentStepItem(item);
        else if (item.id < props.currentStep)
            return passedStepItem(item);
        else if (item.id > props.currentStep)
            return upCommingStepItem(item);
        else return currentStepItem(item);   
    };


    const renderStepItems = () =>{
        return props.stepItems.map((item,index)=>(
            <li key={index} className="relative flex items-center justify-center p-4">
                {stepItem(item)}
            </li>
        ));
    };
 
    return (
        <div>
            <ol className="grid grid-cols-1 divide-x divide-gray-400 overflow-hidden rounded-lg border border-gray-400 text-sm text-gray-500 sm:grid-cols-4">
                {renderStepItems()}
            </ol>
        </div>
    );
};

export default Stepper;
