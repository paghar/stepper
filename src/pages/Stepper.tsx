import {useRouter} from "next/router";
import React from "react";
import {useSelector} from "react-redux";
import {stepItems,stepperNumber} from "../data/constants";
import {numbers} from "../data/enum";
import Stepper from "../components/stepper/Stepper";
import Step1 from "../container/stepper/Step1";
import Step2 from "../container/stepper/Step2";
import Step3 from "../container/stepper/Step3";
import Step4 from "../container/stepper/Step4";

const StepperContainer = () => {

    const usersData = useSelector(state => state as any)?.userState;
    const formStep = usersData.formStep;  
    const router = useRouter(); 
   
    const renderStep = () => {
        if (router.query.formStep === numbers.One)
            return <Step1/>;
        else if (router.query.formStep === numbers.Two)
            return <Step2/>;
        else if (router.query.formStep === numbers.Tree)
            return <Step3/>;
        else if (router.query.formStep === numbers.Four)
            return <Step4/>;
        else 
            return <Step1/>;
    };

    return (
        <div className="m-8">
            <Stepper
                stepprNumber={stepperNumber}
                currentStep={formStep}
                stepItems={stepItems}
            />    
            {renderStep()}       
        </div>       
    );
};

export default StepperContainer;

