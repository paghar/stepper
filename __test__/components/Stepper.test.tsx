import renderer from "react-test-renderer";
import Stepper from "../../src/components/stepper/Stepper";
import {stepItems} from "../../src/data/constants";

describe(">>>> Stepper", () => {
    test("++++ Snapshot of Stepper 1", () => {
        const component = renderer.create(
            <Stepper
                stepprNumber={4}
                currentStep={1}
                stepItems={stepItems}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe(">>>> Stepper", () => {
    test("++++ Snapshot of Stepper 3", () => {
        const component = renderer.create(
            <Stepper
                stepprNumber={4}
                currentStep={3}
                stepItems={stepItems}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

export {};