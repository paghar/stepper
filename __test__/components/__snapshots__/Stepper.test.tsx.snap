// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>> Stepper ++++ Snapshot of Stepper 1`] = `
<div>
  <ol
    className="grid grid-cols-1 divide-x divide-gray-400 overflow-hidden rounded-lg border border-gray-400 text-sm text-gray-500 sm:grid-cols-4"
  >
    <li
      className="relative flex items-center justify-center p-4"
    >
      <span
        className="h-10 w-10  border-2 rounded-full border-blue-600	 text-center text-[14px] font-bold leading-10 text-blue-600"
      >
        1
      </span>
      <span
        className="ml-4 text-base font-medium text-blue-600"
      >
        test text
      </span>
    </li>
  </ol>
</div>
`;

exports[`>>>> Stepper ++++ Snapshot of Stepper 1 1`] = `
<div>
  <ol
    className="grid grid-cols-1 divide-x divide-gray-400 overflow-hidden rounded-lg border border-gray-400 text-sm text-gray-500 sm:grid-cols-4"
  >
    <li
      className="relative flex items-center justify-center p-4"
    >
      <span
        className="h-10 w-10  border-2 rounded-full border-blue-600	 text-center text-[14px] font-bold leading-10 text-blue-600"
      >
        1
      </span>
      <span
        className="ml-4 text-base font-medium text-blue-600"
      >
        Step one
      </span>
    </li>
    <li
      className="relative flex items-center justify-center p-4"
    >
      <span
        className="h-10 w-10  border-2 rounded-full  border-gray-400	 text-center text-[14px] font-bold leading-10 text-bule-600"
      >
        2
      </span>
      <span
        className="ml-4 text-base font-medium text-bule-400"
      >
        Step two
      </span>
    </li>
    <li
      className="relative flex items-center justify-center p-4"
    >
      <span
        className="h-10 w-10  border-2 rounded-full  border-gray-400	 text-center text-[14px] font-bold leading-10 text-bule-600"
      >
        3
      </span>
      <span
        className="ml-4 text-base font-medium text-bule-400"
      >
        Step three
      </span>
    </li>
    <li
      className="relative flex items-center justify-center p-4"
    >
      <span
        className="h-10 w-10  border-2 rounded-full  border-gray-400	 text-center text-[14px] font-bold leading-10 text-bule-600"
      >
        4
      </span>
      <span
        className="ml-4 text-base font-medium text-bule-400"
      >
        Step four
      </span>
    </li>
  </ol>
</div>
`;

exports[`>>>> Stepper ++++ Snapshot of Stepper 3 1`] = `
<div>
  <ol
    className="grid grid-cols-1 divide-x divide-gray-400 overflow-hidden rounded-lg border border-gray-400 text-sm text-gray-500 sm:grid-cols-4"
  >
    <li
      className="relative flex items-center justify-center p-4"
    >
      <span
        className="h-10 w-10 rounded-full bg-blue-600 text-center text-[14px] font-bold leading-10 text-white"
      >
        <svg
          className="h-9 w-9"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <span
        className="ml-4 text-base font-medium text-gray-900"
      >
        Step one
      </span>
    </li>
    <li
      className="relative flex items-center justify-center p-4"
    >
      <span
        className="h-10 w-10 rounded-full bg-blue-600 text-center text-[14px] font-bold leading-10 text-white"
      >
        <svg
          className="h-9 w-9"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <span
        className="ml-4 text-base font-medium text-gray-900"
      >
        Step two
      </span>
    </li>
    <li
      className="relative flex items-center justify-center p-4"
    >
      <span
        className="h-10 w-10  border-2 rounded-full border-blue-600	 text-center text-[14px] font-bold leading-10 text-blue-600"
      >
        3
      </span>
      <span
        className="ml-4 text-base font-medium text-blue-600"
      >
        Step three
      </span>
    </li>
    <li
      className="relative flex items-center justify-center p-4"
    >
      <span
        className="h-10 w-10  border-2 rounded-full  border-gray-400	 text-center text-[14px] font-bold leading-10 text-bule-600"
      >
        4
      </span>
      <span
        className="ml-4 text-base font-medium text-bule-400"
      >
        Step four
      </span>
    </li>
  </ol>
</div>
`;
